### PR TITLE
chore(Checkbox): rewrite tests @testing-library/react

### PR DESCRIPTION
**Pull request by langz**

### CODE REVIEW

File: packages/dnb-eufemia/src/components/checkbox/__tests__/Checkbox.test.tsx [hunk 1]
```diff
@@ -27,40 +27,50 @@ props.label = 'checkbox'
...
+
     // default checked value has to be false
-    expect(Comp.find('input').instance().checked).toBe(false)
+    expect(
+      (screen.getByRole('checkbox') as HTMLInputElement).checked
+    ).toBe(false)
 
-    Comp.find('input').simulate('change') // we could send inn the event data structure like this: , { target: { checked: true } }
-    expect(Comp.find('input').instance().checked).toBe(true)
+    screen.getByRole('checkbox').click()
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/components/checkbox/__tests__/Checkbox.test.tsx:44]: is this the same as `fireEvent.click`?

File: packages/dnb-eufemia/src/components/checkbox/__tests__/Checkbox.test.tsx [hunk 2]
```diff
@@ -79,63 +89,74 @@ describe('Checkbox component', () => {
...
+      expect(
+        (screen.getByRole('checkbox') as HTMLInputElement).checked
+      ).toBe(false)
 
       // set it to true + reset it with null to false
-      Comp.find('button#set-state').simulate('click')
-      Comp.find('button#reset-null').simulate('click')
-      expect(Comp.find('input').instance().checked).toBe(false)
+      screen.getByRole('button', { name: 'set-state' }).click()
+      screen.getByRole('button', { name: 'reset-null' }).click()
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/components/checkbox/__tests__/Checkbox.test.tsx:133]: I would argue, that reading/understanding `document.queryElement('button#set-state')` is more easy than `screen.getByRole('button', { name: 'set-state' })` – but then you probably would need the `fireEvent.click` as well? Like:

```diff
-screen.getByRole('button', { name: 'set-state' }).click()
+fireEvent.click(document.queryElement('button#set-state'))
```

**Review by tujoworker — APPROVED** [review]

File: packages/dnb-eufemia/src/components/checkbox/__tests__/Checkbox.test.tsx (hunk 1 above)
Comment (langz) [reply packages/dnb-eufemia/src/components/checkbox/__tests__/Checkbox.test.tsx:44]: Yes, I believe so 👍

File: packages/dnb-eufemia/src/components/checkbox/__tests__/Checkbox.test.tsx (hunk 2 above)
Comment (langz) [reply packages/dnb-eufemia/src/components/checkbox/__tests__/Checkbox.test.tsx]: Okok, I've now updated these tests to use fireEvent.click and document.querySelector.

**Pull request merged**